depmapomics/upload.py

```diff
@@ -0,0 +1,1298 @@
+                            subset_pr_table.CDSID == latest_cds_id_wes
+                        ].index[0]
+                # if there is wgs, always select wgs
+                else:
+                    subset_seq_table = seq_table[seq_table.index.isin(cds_ids_wgs)]
+                    subset_seq_table.source = subset_seq_table.source.replace(
+                        source_priority
+                    )
+                    latest_cds_id_wgs = subset_seq_table.loc[
+                        cds_ids_wgs, "source"
+                    ].idxmin()
+                    pr = subset_pr_table[
+                        subset_pr_table.CDSID == latest_cds_id_wgs
+                    ].index[0]
+                rows.append((m, pr, "dna"))
+    default_table = pd.DataFrame(rows, columns=["ModelID", "ProfileID", "ProfileType"])
+    return default_table
+
+
+def initVirtualDatasets(
+    samplesetname=SAMPLESETNAME, taiga_folder_id=VIRTUAL_FOLDER, portals=DATASETS
+):
+    """initialize taiga virtual datasets for all 4 portals by uploading an empty dummy file
+    """
+
+    with open("temp/dummy.csv", "w") as fp:
+        pass
+    virutal_pr = dict()
+    virtual_model = dict()
+    tc = TaigaClient()
+    for p in portals:
+        virutal_pr[p] = tc.create_dataset(
+            p + "_" + samplesetname + "_profile",
+            dataset_description=samplesetname
+            + " release of the DepMap dataset for the DepMap Portal. Please look at the README file for additional information about this dataset. ",
+            upload_files=[
+                {
+                    "path": "temp/dummy.csv",
+                    "name": "init",
+                    "format": "Raw",
+                    "encoding": "utf-8",
+                }
+            ],
+            folder_id=taiga_folder_id,
+        )
+        virtual_model[p] = tc.create_dataset(
+            p + "_" + samplesetname + "_model",
+            dataset_description=samplesetname
+            + " release of the DepMap dataset for the DepMap Portal. Please look at the README file for additional information about this dataset. ",
+            upload_files=[
+                {
+                    "path": "temp/dummy.csv",
+                    "name": "init",
+                    "format": "Raw",
+                    "encoding": "utf-8",
+                }
+            ],
+            folder_id=taiga_folder_id,
+        )
+    return virutal_pr, virtual_model
+
+
+def uploadCNMatricesPR(prs, portal, taiga_latest=TAIGA_CN, taiga_virtual=""):
+    """subset, rename, save and upload to taiga PR-level CN matrices
+    
+    Args:
+        prs (list): list of PR-ids to release
+        portal (str): which portal the data is being uploaded to
+        taiga_latest (str): which dataset the matrices to be subsetted are being read from
+        taiga_virtual (str): which dataset the matrices are being uploaded to
+    """
+    folder = "temp/" + portal + "/profile/"
```

Comment:
If this folder is really temp, I'd use tempfile.TemporaryDirectory() but if you want to give it a static name, maybe something other then "temp"

("staging", "working", "output", etc)